trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -1395,30 +1550,35 @@ private Result handleAtomicFetch(final IDispatcher main, final IASTFunctionCallE
 	 *            The C location
 	 * @param body
 	 *            The body that should be atomic based on the memory order
-	 * @param memoryOrder
-	 *            The memory order
+	 * @param memoryOrders
+	 *            The memory orders to apply
 	 * @return An ExpressionResult representing the translation respecting the memory order
 	 */
 	private ExpressionResult applyMemoryOrder(final ILocation loc, final ExpressionResult body,
-			final Expression memoryOrder) {
+			final Expression... memoryOrders) {

```

Comment:
You updated the signature of the function, can you also update the documentation? It is not quite clear to me, how the multiple memory orders are handled here.

Reply:
:+1: hopefully clearer now

Reply:
Thanks!

> If every given memory order is equal to MEMORY_ORDER_SEQ_CST, we just make all statements atomic.

If understand correctly, you create nested ifs:

```
if (mem1 == SEQ_CST) {
  if (mem2 ==  SEQ_CST) {
    atomic { st };
  } else {
    assert false;
} else {
  assert false;
}
```

Wouldn't it be simpler to create simply the conjunction of all memory-order equalities?

```
if (mem1 == SEQ_CST && mem2 ==  SEQ_CST) {
    atomic { st };
} else {
  assert false;
}
```